packages/flutter/lib/src/rendering/mouse_tracking.dart

```diff
@@ -89,6 +169,75 @@ class MouseTrackerAnnotation with Diagnosticable {
       },
       ifEmpty: '<none>',
     ));
+    properties.add(DiagnosticsProperty<PreparedMouseCursor>('cursor', cursor, defaultValue: null));
+  }
+}
+
+class _PlainChangeNotifier extends ChangeNotifier {
+  void notify() {
+    notifyListeners();
+  }
+}
+
+/// A [MouseTrackerAnnotation] that allows varying callbacks and cursors. 
+/// 
+/// The [MouseTrackedRenderObjectMixin] can be used whenever a [MouseTrackerAnnotation]
+/// annotation is needed, since the fields of annotations only matter when it's
+/// searched. However, the [cursorNotifier] must be provided whenever the annotation
+/// is available for searching, and be triggered properly. For more information,
+/// see [MouseTrackerAnnotation.cursorNotifier]
+/// 
+/// See also:
+/// 
+///  * [MouseTracker], which uses [MouseTrackerAnnotation].
+///  * [MouseTrackerAnnotation], which is an immutable implementation of 
+///    this class, and the interface thereof.
+mixin MouseTrackedRenderObjectMixin on RenderObject implements MouseTrackerAnnotation {
+  @override
+  PointerEnterEventListener onEnter;
+
+  @override
+  PointerHoverEventListener onHover;
+
+  @override
+  PointerExitEventListener onExit;
+
+  @override
+  PreparedMouseCursor get cursor => _cursor;
+  PreparedMouseCursor _cursor;
+  set cursor(PreparedMouseCursor value) {
+    final PreparedMouseCursor oldCursor = _cursor;
+    _cursor = value;
+    if (attached && oldCursor != value) {
+      if ((oldCursor != null) == (value != null)) {
+        final _PlainChangeNotifier notifier = _cursorNotifier as _PlainChangeNotifier;
```

Comment:
Just type _cursorNotifier as _PlainChangeNotifier directly?

Reply:
I see this code pattern everywhere (especially in `RenderObject`) that I thought it was not recommended.